tests/integration/targets/ec2_eip/tasks/main.yml

```diff
@@ -7,922 +10,1389 @@
+          - eip.allocation_id is defined and eip.allocation_id.startswith("eipalloc-")
+          - ( eip_info_start.addresses | length ) + 1 == ( eip_info.addresses | length )
+
+    - name: Get EIP info via public ip
+      ec2_eip_info:
+        filters:
+          public-ip: '{{ eip.public_ip }}'
+      register: eip_info
+
+    - assert:
+        that:
+          - '"addresses" in eip_info'
+          - eip_info.addresses | length == 1
+          - eip_info.addresses[0].allocation_id == eip.allocation_id
+          - eip_info.addresses[0].domain == "vpc"
+          - eip_info.addresses[0].public_ip == eip.public_ip
+          - '"AnsibleEIPTestPrefix" in eip_info.addresses[0].tags'
+          - eip_info.addresses[0].tags['AnsibleEIPTestPrefix'] == resource_prefix
+
+    - name: Get EIP info via allocation id
+      ec2_eip_info:
+        filters:
+          allocation-id: '{{ eip.allocation_id }}'
+      register: eip_info
+
+    - assert:
+        that:
+          - '"addresses" in eip_info'
+          - eip_info.addresses | length == 1
+          - eip_info.addresses[0].allocation_id == eip.allocation_id
+          - eip_info.addresses[0].domain == "vpc"
+          - eip_info.addresses[0].public_ip == eip.public_ip
+          - '"AnsibleEIPTestPrefix" in eip_info.addresses[0].tags'
+          - eip_info.addresses[0].tags['AnsibleEIPTestPrefix'] == resource_prefix
+
+    - name: Allocate a new ip (idempotence) - check_mode
+      ec2_eip:
+        state: present
+        public_ip: '{{ eip.public_ip }}'
+      register: eip
+      check_mode: yes
+
+    - assert:
+        that:
+          - eip is not changed
+
+    - name: Allocate a new ip (idempotence)
+      ec2_eip:
+        state: present
+        public_ip: '{{ eip.public_ip }}'
+      register: eip
+
+    - ec2_eip_info:
+      register: eip_info
+
+    - assert:
+        that:
+          - eip is not changed
+          - eip.public_ip is defined and ( eip.public_ip | ansible.utils.ipaddr )
+          - eip.allocation_id is defined and eip.allocation_id.startswith("eipalloc-")
+          - ( eip_info_start.addresses | length ) + 1  == ( eip_info.addresses | length )
+
+    # ------------------------------------------------------------------------------------------
+
+    - name: Release EIP - check_mode
+      ec2_eip:
+        state: absent
+        public_ip: '{{ eip.public_ip }}'
+      register: eip_release
+      check_mode: yes
+
+    - assert:
+        that:
+        - eip_release.changed
+
+    - name: Release eip
+      ec2_eip:
+        state: absent
+        public_ip: '{{ eip.public_ip }}'
+      register: eip_release
+
+    - ec2_eip_info:
+      register: eip_info
+
+    - assert:
+        that:
+          - eip_release.changed
+          - not eip_release.disassociated
+          - eip_release.released
+          - ( eip_info_start.addresses | length ) == ( eip_info.addresses | length )
+
+    - name: Release EIP (idempotence) - check_mode
```

Comment:
Just thinking but probably don't need the `check_mode` tasks for the idempotency tests, typically I think just the 3 cases is enough for each permutation.
 - check_mode
 - live run
 - idempotency check run

Reply:
These are still valuable: in the past some of the modules always returned `true` even when nothing would have changed.